### PR TITLE
docs: benchmark TiDB in Kubernetes

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 

This provides a general way to run sysbench in Kubernetes to benchmark TiDB cluster.

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

**Comment by gregwebs** [comment]
It looks like sysbench is not requesting resources. Users may not choose the dedicated node approach, so I think that is important.

**Comment by gregwebs** [comment]
It looks like this is assuming a root user without a password. The dealing with the password should at least be mentioned in the docs.

**Comment by cofyc** [comment]
> It looks like this is assuming a root user without a password. The dealing with the password should at least be mentioned in the docs.

marked this WIP, I'm going to document how to customize this with kustomize

File: manifests/sysbench/prepare.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        - /bin/sh
+        - -c
+        - |-
+          MYSQL_OPTS="-h${TIDB_HOST} -P${TIDB_PORT} -u${TIDB_USER}"
+          if [ -n "$TIDB_PASSWORD" ]; then
+            MYSQL_OPTS+=" -p${TIDB_PASSWORD}"
+          fi
+          mysql $MYSQL_OPTS -Nse 'set @@global.tidb_hashagg_final_concurrency=1;'
+          mysql $MYSQL_OPTS -Nse 'set @@global.tidb_hashagg_partial_concurrency=1;'
+          mysql $MYSQL_OPTS -Nse 'set @@global.tidb_disable_txn_auto_retry=0;'
```
Comment (gregwebs) [inline manifests/sysbench/prepare.yaml]: This is a side-effecting operation? After sysbench is run the configuration will be changed.
The approach I try to take is to alter the TiDB configuration to the desired production settings and then to not make any special configuration changes for the purpose of the benchmark.

Comment (cofyc) [reply manifests/sysbench/prepare.yaml]: I guess it is, it is taken from this benchmark report: https://pingcap.com/docs/v3.0/benchmark/sysbench-v4/

it seems not good to alter database in these general benchmark jobs, I'll remove it

Comment (gregwebs) [reply manifests/sysbench/prepare.yaml]: The documentation here can point to advice for configuration. It is also possible to add an opt-in option to set this configuration.

File: docs/BenchmarkTiDB.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Benchmark TiDB
```
Comment (tennix) [inline docs/BenchmarkTiDB.md]: It's better to rename it as `benchmark-tidb.md`

File: manifests/sysbench/bench.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  parallelism: 3
+  template:
+    metadata:
+      labels:
+        app: sysbench-bench
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: sysbench
+        image: uhub.ucloud.cn/pingcap/sysbench:latest
```
Comment (tennix) [inline manifests/sysbench/bench.yaml]: Can this image be customized?

File: docs/BenchmarkTiDB.md (hunk 2 above)
Comment (cofyc) [reply docs/BenchmarkTiDB.md]: done

File: manifests/sysbench/bench.yaml (hunk 3 above)
Comment (cofyc) [reply manifests/sysbench/bench.yaml]: yes, add a section "More customizations" for this

**Pull request merged**